### PR TITLE
runtime: Document SET_AUTH_MANIFEST and AUTHORIZE_AND_STASH CMDs

### PR DESCRIPTION
**Pull request by ArthurHeymans**

### CODE REVIEW

File: api/src/mailbox.rs [hunk 1]
```diff
@@ -1024,7 +1024,6 @@ impl AuthAndStashFlags {
 #[derive(Debug, AsBytes, FromBytes, PartialEq, Eq)]
 pub struct AuthorizeAndStashReq {
     pub hdr: MailboxReqHeader,
-    pub metadata: [u8; 4],
```
Comment (jhand2) [inline api/src/mailbox.rs:1027]: metadata is included to be used as the DPE type. We should leave this.

File: runtime/README.md [hunk 2]
```diff
@@ -711,8 +711,83 @@ Command Code: `0x434B_4558` ("CKEX")
...
+| chksum        | u32      | Checksum over other output arguments, computed by Caliptra. Little endian.
+| fips\_status  | u32      | Indicates if the command is FIPS approved or an error.
+
 
+### AUTHORIZE\_AND\_STASH
+
+Command Code: `0x4154_5348` ("ATSH")
+
+*Table: `AUTHORIZE\_AND\_STASH` input arguments*
+| **Name**    | **Type** | **Description**                                                             |
```
Comment (jhand2) [inline runtime/README.md:771]: The regular STASH_MEASUREMENT call has a `context` field:

|  |  | |
| -- | -- | --| 
| context | u8[48] | Context field for svn; e.g., a hash of the public key that authenticated the SVN. |

This is used in the stable identity proposal.

I think we should include that in this command as well.

File: runtime/README.md [hunk 3]
```diff
@@ -711,8 +711,83 @@ Command Code: `0x434B_4558` ("CKEX")
...
+*Table: `SET\_AUTH\_MANIFEST` input arguments*
+
+| **Name**                      | **Type**     | **Description**                                                             |
+| chksum                        | u32          | Checksum over other input arguments, computed by the caller. Little endian. |
+| manifest size                 | u32          | The size of the full Authentification Manifest                              |
+| preamble\_marker              | u32          | Marker needs to be 0x4154_4D4E for the preamble to be valid                 |
+| preamble\_size                | u32          | Size of the preamble                                                        |
+| preamble\_version             | u32          | Version of the preamble                                                     |
+| preamble\_flags               | u32          | Preamble flags                                                              |
+| preamble\_vendor\_ecc\_key    | u32[24]      | Vendor ECC key with R and S coordinates in that order                       |
```
Comment (jhand2) [inline runtime/README.md:727]: Do you mean X and Y?

File: runtime/README.md [hunk 4]
```diff
@@ -711,8 +711,83 @@ Command Code: `0x434B_4558` ("CKEX")
...
+| preamble\_vendor\_lms\_key    | u32[6]       | Vendor LMS key                                                              |
+| preamble\_vendor\_ecc\_sig    | u32[24]      | Vendor LMS signature                                                        |
+| preamble\_vendor\_LMS\_sig    | u32[1344]    | Vendor LMS signature                                                        |
+| preamble\_vendor\_ecc\_key    | u32[24]      | Owner ECC key with R and S coordinates in that order                        |
+| preamble\_vendor\_lms\_key    | u32[6]       | Owner LMS key                                                               |
+| preamble\_vendor\_ecc\_sig    | u32[24]      | Owner LMS signature                                                         |
+| preamble\_vendor\_LMS\_sig    | u32[1344]    | Owner LMS signature                                                         |
```
Comment (jhand2) [inline runtime/README.md:734]: Field names for this should be `*_owner_*`

File: runtime/README.md [hunk 5]
```diff
@@ -711,8 +711,83 @@ Command Code: `0x434B_4558` ("CKEX")
...
+| metadata\_vendor\_ecc\_sig    | u32[24]      | Metadata Vendor LMS signature                                               |
+| metadata\_vendor\_LMS\_sig    | u32[1344]    | Metadata Vendor LMS signature                                               |
+| metadata\_owner\_ecc\_sig     | u32[24]      | Metadata Owner LMS signature                                                |
+| metadata\_owner\_LMS\_sig     | u32[1344]    | Metadata Owner LMS signature                                                |
```
Comment (jhand2) [inline runtime/README.md:738]: Update descriptions to match the algorithm in the field name

File: runtime/README.md [hunk 6]
```diff
@@ -711,8 +711,83 @@ Command Code: `0x434B_4558` ("CKEX")
...
-## Checksum
+### SET\_AUTH\_MANIFEST
+
+Command Code: `0x4154_4D4E` ("ATMN")
+
+*Table: `SET\_AUTH\_MANIFEST` input arguments*
+
+| **Name**                      | **Type**     | **Description**                                                             |
+| chksum                        | u32          | Checksum over other input arguments, computed by the caller. Little endian. |
+| manifest size                 | u32          | The size of the full Authentification Manifest                              |
```
Comment (jhand2) [inline runtime/README.md:722]:
```suggestion
| manifest size                 | u32          | The size of the full Authentication Manifest                              |
```

File: runtime/README.md [hunk 7]
```diff
@@ -711,8 +711,83 @@ Command Code: `0x434B_4558` ("CKEX")
 | fips\_status        | u32       | Indicates if the command is FIPS approved or an error.
 | certify\_key\_resp  | u8[2176]  | Certify Key Response.
 
-## Checksum
+### SET\_AUTH\_MANIFEST
```
Comment (jhand2) [inline runtime/README.md:714]: Is there a document section we can link to here with the manifest format and general design?

File: api/src/mailbox.rs (hunk 1 above)
Comment (ArthurHeymans) [reply api/src/mailbox.rs:1027]: Should the code use it somehow?

File: runtime/README.md (hunk 7 above)
Comment (mhatrevi) [reply runtime/README.md:714]: We have the design document on the shared drive; will that suffice, or should we update the Runtime README.md? https://docs.google.com/document/d/1BofLEAzeszgAGVcaEjTxM7tDxwIq9KhVXuqGUlrg1sc/edit#heading=h.z8ehugjyw5jp

File: api/src/mailbox.rs (hunk 1 above)
Comment (jhand2) [reply api/src/mailbox.rs:1027]: Yes, it should be used the same way it's used for STASH_MEASUREMENT: https://github.com/chipsalliance/caliptra-sw/blob/main/runtime/src/stash_measurement.rs#L80

File: runtime/README.md (hunk 7 above)
Comment (jhand2) [reply runtime/README.md:714]: That's fine for now. We can move it to markdown later.

File: api/src/mailbox.rs (hunk 1 above)
Comment (ArthurHeymans) [reply api/src/mailbox.rs:1027]: The stashing is marked as TODO I see

**Pull request merged**